mysql-test/t/check_constraint.test

```diff
@@ -176,3 +176,16 @@ select * from t1 where a is null;
 --error ER_CONSTRAINT_FAILED
 insert t1 (b) values (1);
 drop table t1;
+
+--echo #
+--echo # MDEV-18463 Don't allow multiple table CONSTRAINTs with the same name
+--echo #
+
+create table tpk
+(id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,name VARCHAR(100) NOT NULL);
+--error ER_DUP_CONSTRAINT_NAME
+create table tfk
+(id INT, c1 INT, c2 INT NOT NULL,
+ CONSTRAINT sid FOREIGN KEY (`c1`) REFERENCES tpk (`id`)
+ ON DELETE NO ACTION ON UPDATE NO ACTION, CONSTRAINT sid check (c2>15));
+drop table tpk;
```

Comment:
I would suggest more tests, and I think that we need a variant of the test where both tables are created with `ENGINE=InnoDB` in order to actually have the `FOREIGN KEY` constraints created:
```sql
CREATE TABLE tpk(id INT PRIMARY KEY);
CREATE TABLE tfk (c1 INT, c2 INT, CONSTRAINT sid FOREIGN KEY (c1) REFERENCES tpk (id));
--error ER_DUP_CONSTRAINT_NAME
ALTER TABLE tfk ADD CONSTRAINT Sid CHECK (c2>15);
ALTER TABLE tfk DROP FOREIGN KEY sid;
ALTER TABLE tfk ADD CONSTRAINT Sid CHECK (c2>15);
--error ER_DUP_CONSTRAINT_NAME
ALTER TABLE tfk ADD CONSTRAINT sid FOREIGN KEY (c1) REFERENCES tpk (id);
--error ER_DUP_CONSTRAINT_NAME
ALTER TABLE tfk ADD CONSTRAINT FOREIGN KEY sid (c1) REFERENCES tpk (id);
ALTER TABLE tfk ADD CONSTRAINT tid FOREIGN KEY (c1) REFERENCES tpk (id);
DROP TABLE tfk, tpk;
```
In `main.foreign_key`, I see no `ER_DUP_CONSTRAINT_NAME`. I would suggest to add a test for that too:
```sql
CREATE TABLE tfk (c1 INT, c2 INT, CONSTRAINT sid FOREIGN KEY (c1) REFERENCES tpk (id));
--error ER_DUP_CONSTRAINT_NAME
ALTER TABLE tfk ADD CONSTRAINT sid FOREIGN KEY (c2) REFERENCES tpk(id);
DROP TABLE tfk,tpk;
```

Reply:
Hi @dr-m,I have updated tests.

The last test case is showing the field check constraint vs foreign key constraint with the same name. Currently it is not covered so I'm not sure should be covered at all. 
Please let me know should this be changed also.
Thanks